hack/common.sh

```diff
@@ -103,10 +103,23 @@ sti::build::build_binaries() {
     for platform in "${platforms[@]}"; do
       sti::build::set_platform_envs "${platform}"
       echo "++ Building go targets for ${platform}:" "${targets[@]}"
-      go install "${goflags[@]:+${goflags[@]}}" \
-          -ldflags "${version_ldflags}" \
-          "${binaries[@]}"
-      sti::build::unset_platform_envs "${platform}"
+      # have to enable these debug flags for using gdb separately from goflags
+      # because the -gcflags "-N -l" ... i.e. the -N and -l are the setting for -gcflags
+      # not for go itself .. for now we punt on other goflags if developers are
+      # compiling to run with gdb
+      if [[ -v STI_ENABLE_GDB ]]; then
+	  echo "Enabling compiling go with debug to allow gdb usage (i.e. -gcflags \"-N -l\") since STI_ENABLE_GDB is set"
+	  echo "In this case, STI_ENABLE_GDB is set to  ${STI_ENABLE_GDB}"
+	  go install -gcflags "-N -l"\
+             -ldflags "${version_ldflags}" \
+             "${binaries[@]}"
+	  sti::build::unset_platform_envs "${platform}"
+      else
+	  go install "${goflags[@]:+${goflags[@]}}" \
+             -ldflags "${version_ldflags}" \
+             "${binaries[@]}"
+	  sti::build::unset_platform_envs "${platform}"
+      fi
```

Comment:
Setting of STI_GOFLAGS unfortuanately did NOT work as I explained in the above comment.  Thought in looking at that comment, I see I failed to mention STI_GOFLAGS ;-) ... I'll update the comment.

My intent here was to ultimately update origin as well, but I wanted to try some more things within the less active source-to-image repository first.  As it is turning out, I'm having to do my gdb investigation in small spurts as I get new assignments (which still invariably becomes educational and take some time), but wanted to avoid having to copy in and copy out my local copy and hack/common.sh as I found time to revisit gdb.

If I add the additional disclaimer that this will either be a) duplicated in origin as time permits, or b) will remove if gdb proves non-useful for Go dev, are we good then with leaving this change in?